### ISSUE TITLE
One year later. Still no success.

### ISSUE THREAD

**Issue by GermanBluefox** [actions-on-google/smart-home-nodejs#176]
Is it really so hard to get the things running?

As described in https://github.com/actions-on-google/smart-home-nodejs/issues/50 I tried today anew the action and still no luck.

I created the server https://iobroker.live

![image](https://user-images.githubusercontent.com/4582016/46904938-93ad1e00-ceec-11e8-93b2-79a30377483c.png)

I created the action and I can see it in the list of action in google home app.
![image](https://user-images.githubusercontent.com/4582016/46904953-d838b980-ceec-11e8-876b-9060cac1737d.png)

The authentication process is OK
![image](https://user-images.githubusercontent.com/4582016/46904974-28b01700-ceed-11e8-8d0a-a00aab254328.png)

But then:
![image](https://user-images.githubusercontent.com/4582016/46904980-41203180-ceed-11e8-8a8d-26fa53be494a.png)
It is like: 'The configuration cannot be updated. Check your connection.'

Almost same error as one year before.
I still can't get the first SYNC request!

Please HELP!!!!!!

**Comment by GermanBluefox** [comment]
Google HomeGraph API is enabled too.

![image](https://user-images.githubusercontent.com/4582016/46905135-aa08a900-ceef-11e8-9a72-5491cb5e7e96.png)

**Comment by GermanBluefox** [comment]
HomeGraph answers only 404

**Comment by madgeni** [comment]
Think that will fail unless you've linked your account ok - so when you link, does it go through the oauth handshake ok?

**Comment by GermanBluefox** [comment]
Oauth is completely OK. I can see how access and refresh token are generated

**Comment by Fleker** [comment]
In the Cloud Console, you should see a link for [Stackdriver](https://console.cloud.google.com/logs/viewer). This is where you should see [error logging](https://developers.google.com/actions/smarthome/logging) for your Action. An error is likely happening between the OAuth and Sync events. You may need to change the filtering to see all errors.

Until the first SYNC, HomeGraph APIs will not work as you'd be trying to request or send data for non-existent users.

If nothing in the logs seems to indicate an error, we can try to look deeper into your project or account settings which may show a more specific issue that you're experiencing.

**Comment by GermanBluefox** [comment]
> In the Cloud Console, you should see a link for [Stackdriver](https://console.cloud.google.com/logs/viewer). This is where you should see [error logging](https://developers.google.com/actions/smarthome/logging) for your Action. An error is likely happening between the OAuth and Sync events. You may need to change the filtering to see all errors.
> 
> Until the first SYNC, HomeGraph APIs will not work as you'd be trying to request or send data for non-existent users.
> 
> If nothing in the logs seems to indicate an error, we can try to look deeper into your project or account settings which may show a more specific issue that you're experiencing.

Which topic should I check?
![image](https://user-images.githubusercontent.com/4582016/46976573-27881100-d0ca-11e8-80af-0fa5fbfabbaa.png)

I must say, that I checked all of them and could not find any logs. Only provisioning logs are there.

**Comment by Fleker** [comment]
I would check **Google Assistant Action** and **Google Project**. If you do not see anything, I'd like to be able to get the project shared with me to see if I can reproduce the error on the project or if the issue may be related to something else.

**Comment by GermanBluefox** [comment]
I granted you the viewer access.

**Comment by Fleker** [comment]
Can you send me an email with the project id?

**Comment by GermanBluefox** [comment]
Sent

ср, 17 окт. 2018 г. в 20:28, Nick <notifications@github.com>:

**Comment by smashah** [comment]
This is happening because the code is playing hot potato with the client secret and ID and actions by google isn't forwarding that data to each subsequent request. I don't know what's going on on googles end but I have a fix. What you need to do is open up auth-provider.js and hard code clientId & clientSecret and get rid of all instances of: 

`
  let clientId = req.query.client_id
      ? req.query.client_id : req.body.client_id;
  let clientSecret = req.query.client_secret
      ? req.query.client_secret : req.body.client_secret;`

**Comment by smashah** [comment]
#186

**Comment by GermanBluefox** [comment]
@Fleker 
Has something changed? 
I tried now (without any code modifications) the linking again and I HAVE GOT **SYNC** REQUEST!!! :)

**Comment by GermanBluefox** [comment]
@Fleker 
And again no luck:
- I logged in on the web page for my smarthome. I could not see any devices there, so I created a new one.
- I deleted the linking and started the link anew.
- NO linking there.

That mean someone created the devices and I could read them, but if I create the devices on my own it does not work.

**Comment by GermanBluefox** [comment]
I could see, that the state in redirect is different.
- For me it is like 
![image](https://user-images.githubusercontent.com/4582016/47682882-18c35300-dbce-11e8-93e0-2bc480474c3c.png)
I could see some text.

- And working redirect looks like:
![image](https://user-images.githubusercontent.com/4582016/47682916-31336d80-dbce-11e8-87cf-539f681e7074.png)
No text at all and longer.

**Comment by Fleker** [comment]
I had managed to sync your project without issue. We're investigating why this is failing in certain cases.

**Comment by DutchmanNL** [comment]
> I had managed to sync your project without issue. We're investigating why this is failing in certain cases.

Hi Fleker, would be great if you can provide us an status update we really would like to implement googlehome support to our smarthome engine.

**Comment by johannes-keinestam** [comment]
Don't know if this helps debugging, but it seems like it works for me now when I deploy my code to AWS instead of running it locally through ngrok or localtunnel. Just wildly guessing here -- but since (at least on iOS) the oauth flow is done through a browser and works just fine, and presumably the SYNC request is being sent by the Google Home/Assistant app itself and not the browser, could it be some sort of certificate or DNS issue with the app?

**Comment by GermanBluefox** [comment]
Any news here? I cant open the issue on official support, because they sad, that I will be helped here...

**Comment by TA2k** [comment]
@Fleker Do you have any updates regarding the SYNC problem. Because a large amount of the  iobroker smarthome community is waiting on this ticket to be resolved to integrate their google homes into the iobroker like other smart speaker systems.

**Comment by DutchmanNL** [comment]
@johannes-keinestam @GermanBluefox 

Is there anything we can make some progress togehter in this matter ?
I Understand from Johannes he tried the project code on an AWS instance succesfully, doe that mean our way of interaction with the API is not accepted or faulty ?

What can be the next step in here for troubleshooting and making progress i have the feeling we run in a deadlock since a year ;)

**Comment by Elle04** [comment]
Please Google HELP!

**Comment by johannes-keinestam** [comment]
@DutchmanNL I am actually not running this project specifically, but my own serverless/nodejs implementation modelled after the documentation found at [1], so I can't guarantee that following my steps would solve the issue. However, since the issue seems to stem simply from the Google Home app not being able to even perform a HTTP request against some arbitrary URL, I would imagine this issue can occur for any implementation of this protocol.

Deploying the server to some sort of web hosting might solve the problem, and I would recommend at least trying it. Does your product require the Google Home code to be deployed locally at each user's computer, or are you just running it locally for testing purposes? If it's the latter, I would suggest actually testing deployed to the cloud, even though it is definitely more cumbersome...

[1] https://developers.google.com/actions/smarthome

**Comment by LovelySunshine** [comment]
why this stuff does not work? 
Google home is not working, nothing happens.  
It didn't work.  

Can GOOGLE please help me.  

I am currently having the same problem

Is there any solution?

Thanks

**Comment by wendy2702** [comment]
Hi Google, can you please start to support this?

I owned Google Home instead of Amazon Echo but it looks it was a failure to trust on Google .

Up to know my Google Home is useless... :-(

**Comment by emddummi** [comment]
Hi Google, can you please start to support this?

I owned Google Home instead of Amazon Echo but it looks it was a failure to trust on Google .

Up to know my Google Home is useless... :-(

**Comment by OlegJktu** [comment]
https://www.home-assistant.io/components/google_assistant/#troubleshooting-with-nginx

**Comment by manitwo08** [comment]
Come on Google.
Pls support this. We really want it.

**Comment by Jeeperat** [comment]
Hi Google,
Please help us to get this Feature

**Comment by Maja61** [comment]
Please help us to fix it!

**Comment by Tobi-BPi** [comment]
Come on Google. 

Two of your Google home will start to work in my home.... 

Please support this!

**Comment by BluePixelHD** [comment]
Google, please help us to fix it!

**Comment by TomT1965** [comment]
I think the problem should be fixed fast....
Or do all ioBroker users buy Alexa insteadt and trash Google home forever?

**Comment by Brainbug01** [comment]
Please help us to fix it!

**Comment by eric1905** [comment]
Would be great if this will be fixed very soon :)

**Comment by PlhsBeatz** [comment]
Google, pls fix this.

**Comment by salambelgord** [comment]
Hey Google, help us

**Comment by Real4el** [comment]
OK Google, please help us to fix it!

**Comment by JayJayPott** [comment]
OK Google, please help us to fix it!

**Comment by si021012** [comment]
A fix would be in the interest of thousands of users!

**Comment by IO-Link** [comment]
Google - please help! I don't to switch to Amazons Alexa...

**Comment by quorle** [comment]
Please Google help to get this fixed!

**Comment by BassTii** [comment]
Google, please help us to fix it!

**Comment by uklatt** [comment]
Hey Google, please fix this ioBroker problem. I want to use Google Home with IoBroker.
Uwe

**Comment by Kunibert-007** [comment]
Google, what's going on!? Please fix ... Or do you want I buy Alexa?

**Comment by coyote1** [comment]
Google, please help us to fix it!

**Comment by tombenzeen** [comment]
Google, please fix...

**Comment by siggi85** [comment]
Google please solve this problem. Would love to use my google homes with ioBroker!

**Comment by GermanBluefox** [comment]
I created a completely new account (one more time) and now, hallelujah, I could see the created devices.... but my own account still does not work. Very strange.

**Comment by Maranja** [comment]
Please Google, we want to use the Google home with ioBroker!

**Comment by Xyolyp** [comment]
Ok Google, please fix the issue that we can finally use the google home with ioBroker.

**Comment by saepfle** [comment]
Hey Google - common move your a** and fix it !!!

**Comment by Fleker** [comment]
Hey folks,
We're going to take a deep look at this starting on Monday. In the meantime, I hope everyone had a pleasant Thanksgiving and I wish everyone has a good weekend.

**Comment by dabbeljuh22** [comment]
Many thanks @Fleker, we would appreciate to use Google instead of Alexa in our Smarthomes soon!!! Have a great weekend, too.

**Comment by sphings79** [comment]
Thx @Fleker. Hope i can use my GH soon with ioBroker :)

**Comment by DutchmanNL** [comment]
@Fleker, great thank you so much as you see there is a quite big community waiting just to kick off HeyGoogle and decommission these Alexas

**Comment by GermanBluefox** [comment]
As I wrote, the communication with OTHER account now works and the demo does not run now.
If you need it for Investigation I can restore all settings, because demo server still runs.

I have now other Problem, that I can't rename the invocation name
![enums](https://user-images.githubusercontent.com/4582016/48998203-0b5ca280-f153-11e8-8293-1d3c10d99b68.gif)
And no errors are shown.

**Comment by snador1977** [comment]
Thanks @Fleker for your work. We are all waiting for Alexa to be replaced with the Assitant and a huge thank you to BlueFox who, despite all the difficulties with the adapter, stays tuned and continues with the work! :-)

**Comment by Fleker** [comment]
@GermanBluefox there may be some restrictions around display names with one word, as you will first need to do [brand verification](https://developers.google.com/actions/console/connecting-properties). The invocation name won't matter a lot except for the display page. I'll find someone to track down the lack of errors here.

We're continuing to look into the original OAuth failure. Was the email you were using for development a GSuite account, or a general x@gmail.com?

**Comment by GermanBluefox** [comment]
> @GermanBluefox there may be some restrictions around display names with one word, as you will first need to do [brand verification](https://developers.google.com/actions/console/connecting-properties). The invocation name won't matter a lot except for the display page. I'll find someone to track down the lack of errors here.
> 

I could set this name for English and Russian
![image](https://user-images.githubusercontent.com/4582016/49112880-41557000-f294-11e8-8362-dfb181b016f9.png)

> We're continuing to look into the original OAuth failure. Was the email you were using for development a GSuite account, or a general [x@gmail.com](mailto:x@gmail.com)?
It was just normal x@gmail.com

**Comment by eric1905** [comment]
@Fleker is there any new information?

**Comment by Fleker** [comment]
It seems like the ioBroker integration is coming along, now unimpeded.

We're still trying to figure out how, but it seemed like the OP's user account was unable to fully place the project into a test state, although others were able to put it into a test state without issue.

**Comment by sphings79** [comment]
@GermanBluefox any news?

**Comment by LovelySunshine** [comment]
Google i am so disappointed... already another 2 month... after ONE year of waiting... and still nothing works!

**Comment by b3ta1978** [comment]
It's a shame that the support always keeps it that way. If it is supposed to work drann should not always pass weeks until the next episode.

**Comment by Fleker** [comment]
I think that the original issue in question has come to a resolution, so this issue will be closed. If anyone else is having a weird issue with authorizing their project, please file a new issue and describe your steps in a way that can be reproduced. Thanks!

**Comment by ChrSch84** [comment]
Any News from Google??

**Comment by Fleker** [comment]
This thread has become off-topic.
